doc/quick_start/contributor-guide.rst

```diff
@@ -0,0 +1,277 @@
+we want to avoid duplicating such minor issues in the Ostro Project JIRA and instead the but should be reported
+to the upstream projects's issuse tracking system.
+
+When an bug is discovered:
+
+#. Find the proper Ostro OS component for tracking the bug:
+     * If the bug is caused by Ostro OS code, an Ostro Project feature or component owner should address it. (Ostro Project
+       components and the responsible owner can be found on the "Components" page in `Ostro Project JIRA`_.)
+     * If the bug is caused by upstream code, the feature owner and QA owner should report in the Ostro OS JIRA and 
+       map it to an upstream project’s bug tracking system. 
+
+   Similarly for feature requests, submit them to the appropriate component or upstream project's issue tracking system.
+
+#. Once the bug is fixed upstream, the feature owner owns merging the upstream code back into Ostro OS. 
+#. QA will verify the fix with the Ostro image and update the Ostro Project bug status properly, and track the issue until it's closed.
+
+The upstream developer and QA owner take responsibility to drive a fix. Based on the bug's impact and available resources, 
+the Ostro feature owner decides whether a hotfix is required in Ostro OS before the upstream project releases a fixed version. The
+feature owner is also responsible for removing the hotfix when appropriate.
+
+When submitting a new bug report (or feature request) to Ostro Project or an upstream project, verify 
+that the issue has not been reported already to avoid duplication. 
+
+
+Submitting Patches to the Ostro Project
+=======================================
+
+For general patches to upstream packages, we recommend you submit them directly to the
+appropriate upstream project home. For patches specific to the Ostro OS, submit them as a 
+git pull request (PR) so project maintainers can review and merge them. In this section we'll 
+explain how to properly format and submit your patch.
+
+In a collaborative open source environment, standards and methods for submitting changes 
+help reduce chaos that can result from an active development community. One general practice 
+is to make small, controlled changes. This practice simplifies review, makes merging and 
+rebasing easier, and keeps the change history clear and clean.
+
+The Ostro OS is based on Yocto Project layers that are put together using the combo-layer script.
+The http://github.com/ostroproject/ostro-os repository is a combination of several components in a 
+single repository and contains everything needed to build Ostro OS, including:
+
+- bitbake
+- openembedded-core
+- meta-intel
+- meta-ostro
+- meta-ostro-fixes
+- meta-ostro-bsp
+- meta-intel-iot-security
+- meta-appfw
+- meta-openembedded
+- meta-oic
+- meta-intel-iot-middleware
+- meta-iotqa
+- meta-iot-web
+- meta-security-isafw
+- meta-yocto
+- meta-java
+- meta-soletta
+- meta-swupd
+
```

Comment:
Keeping these list synchronized is error-prone. Can we just give some examples here and instead refer to the README for the full list?